source/extensions/common/wasm/context.h

```diff
@@ -406,6 +419,80 @@ class Context : public Logger::Loggable<Logger::Id::wasm>,
   uint32_t status_code_{0};
   absl::string_view status_message_;
 
+  Filters::Common::Expr::Activation activation_{};
+
+  // Stream info bound to either a request or a connection.
+  const StreamInfo::StreamInfo* stream_info_{};
+  void setStreamInfo(const StreamInfo::StreamInfo& stream_info) {
+    if (&stream_info != stream_info_) {
+      stream_info_ = &stream_info;
+      // note: none of these use an arena to produce a wrapper, which means
+      // they are safe to re-use across evaluations
+      activation_.InsertValueProducer(
+          Filters::Common::Expr::Connection,
+          std::make_unique<Filters::Common::Expr::ConnectionWrapper>(stream_info));
```

Comment:
Why are we adding the same stream_info five times?

Reply:
These are different wrappers around the same info object. Agree, it's not elegant. I think we would want to have something like getProperty in Activation itself. But that's a change I have to pull through google3, Envoy, etc.